tests/gordo_components/cli/test_cli.py

```diff
@@ -333,24 +332,37 @@ def test_build_cv_mode_build_only(tmp_dir: tempfile.TemporaryDirectory):
             assert metadata_json["model"]["cross-validation"]["scores"] == {}
 
 
-@pytest.mark.parametrize("host", (None, "127.0.0.0"))
-@pytest.mark.parametrize("port", (None, 1234))
-def test_start_server_cli(host, port):
+@pytest.mark.parametrize(
+    "arg,value,exception_expected",
+    [
+        # Valid values
+        ("--host", "0.0.0.0", False),
+        ("--host", "127.0.0.0", False),
+        ("--port", 5555, False),
+        ("--workers", 1, False),
+        ("--log-level", "info", False),
+        ("--log-level", "debug", False),
+        # Invalid values
+        ("--host", "0.0.0", True),
+        ("--port", 0, True),
+        ("--port", 70000, True),
+        ("--workers", -1, True),
+        ("--log-level", "badlevel", True),
+    ],
+)
+def test_gunicorn_execution_hosts(monkeypatch, arg, value, exception_expected):
+    """
+    Test the validation of input parameters to the `run_server` function via the gordo cli
+    """
+
     runner = CliRunner()
-    with mock.patch(
-        "gordo_components.server.server.run_server", mock.MagicMock(return_value=None)
-    ) as m:
-        args = ["run-server"]
-
-        # Add any optional params
-        if host is not None:
-            args.extend(["--host", host])
-        if port is not None:
-            args.extend(["--port", str(port)])
-        result = runner.invoke(cli.gordo, args)
-
-        assert result.exit_code == 0
-        m.assert_called_once_with(host, port)
+    args = ["run-server", arg, value]
+
+    monkeypatch.setattr(
```

Comment:
do you think this should have something like `patched.assert_called_with(args)` or something? Not a big deal I suppose, but won't catch the fail if someone just deletes `subprocess.check_call` all together.

Reply:
Hmmm... good point. I'll use `assert_called_with` then.